antarest/study/storage/utils.py

```diff
@@ -383,3 +384,48 @@ def ignore_outputs(directory: str, _: t.Sequence[str]) -> t.Sequence[str]:
         study.tree.denormalize()
         duration = "{:.3f}".format(time.time() - stop_time)
         logger.info(f"Study '{study_dir}' denormalized in {duration}s")
+
+
+def is_folder_safe(workspace: WorkspaceConfig, folder: str) -> bool:
+    """
+    Check if the provided folder path is safe to prevent path traversal attack.
+
+    Args:
+        workspace: The workspace name.
+        folder: The folder path.
+
+    Returns:
+        `True` if the folder path is safe, `False` otherwise.
+    """
+    requested_path = workspace.path / folder
+    safe_dir = os.path.realpath(workspace.path)
+    real_requested_path = os.path.realpath(requested_path)  # resolve symbolic links like ~ , .. and .
+    common_prefix = os.path.commonprefix(
+        (os.path.realpath(real_requested_path), safe_dir)
+    )  # get the common prefix of the two paths
+    return common_prefix == safe_dir  # if requested path isn't in workspace, then common prefix is different
+
+
+def is_study_folder(path: Path) -> bool:
+    return path.is_dir() and (path / "study.antares").exists()
+
+
+def get_workspace_from_config(config: Config, workspace_name: str, default_allowed: bool = False) -> WorkspaceConfig:
+    if not workspace_name:
+        raise ValueError("workspace_name must be specified")
```

Comment:
Don't think you need to perform this check you'll enter the KeyError which isn't a bad thing